modules/auxiliary/gather/zoomeye_search.rb

```diff
@@ -109,25 +112,25 @@ def match_records?(records)
   end
 
   def run
-    dork = datastore['ZOOMEYE_DORK']
+    dork = datastore['QUERY']
     resource = datastore['RESOURCE']
     maxpage = datastore['MAXPAGE']
     facets = datastore['FACETS']
     api_key = datastore['ZOOMEYE_APIKEY']
     # check to ensure api.zoomeye.org is resolvable
     unless zoomeye_resolvable?
-      print_error('Unable to resolve api.zoomeye.hk')
+      print_error('Unable to resolve api.zoomeye.ai')
       return
     end
 
     first_page = 0
     results = []
-    results[first_page] = dork_search(resource, dork, 1, facets, api_key)
+    results[0] = dork_search(resource, dork, 1, facets, api_key)
 
-    if results[first_page].nil? || results[first_page]['total'].nil? || results[first_page]['total'] == 0
+    if results[0]['total'].nil? || results[0]['total'] == 0
       msg = 'No results.'
-      if !results[first_page]['error'].to_s.empty?
-        msg << " Error: #{results[first_page]['error']}"
+      if results[first_page]['error'].to_s.length > 0
```

Comment:
Rails includes handy `#present?` and `#blank?` matches that make this a bit more readable.
```suggestion
      if results[first_page]['error'].present?
```